### PR TITLE
More options and some cleanup

### PR DESCRIPTION
**Pull request by spacetc62**
* add options like batchSize/noCursorTimeout/maxTimeMS to various db calls
* remove packages `defaults` and `assign` and use modern node.js
* rest of files are now all linted
* upgrade bson/mongodb/object-path
* fix issue where read pref was not applied for counts
* remove some extraneous debug calls

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -271,26 +276,21 @@ class Database extends EventEmitter {
...
    * @see {@link https://mongodb.github.io/node-mongodb-native/4.0/classes/collection.html#updateOne}
+   * @see {@link https://mongodb.github.io/node-mongodb-native/4.0/interfaces/updateoptions.html#writeconcern}
    */
-  async update(collectionName, query, update, options) {
-    options = defaults(options, {
-      multiple: false,
-      upsert: false
-    });
-    debug('getting db collection');
+  async update(collectionName, query, update, options = {}) {
```
Comment (eeerin) [inline lib/db.js:286]: default multiple and upsert to false?

Comment (spacetc62) [reply lib/db.js:286]: was just letting undefined be falsy - not a fan?

File: lib/model.js [hunk 2]
```diff
@@ -312,7 +299,7 @@ const _create = (data, definition) => {
     debug(`remove from ${collection} successful`);
     if ('function' === typeof this.afterRemove) {
       const afterRemove = this.afterRemove(options);
-      if (options.waitAfterRemove) { await afterRemove; }
+      if (options?.waitAfterRemove) { await afterRemove; }
```
Comment (eeerin) [inline lib/model.js:302]: you could default options = {} instead of using ?

Comment (spacetc62) [reply lib/model.js]: done

**Review by eeerin — APPROVED** [review]

**Pull request merged**